### ISSUE TITLE
bring test coverage to 100%

### ISSUE THREAD

**Issue by rien** [dodona-edu/dodona#1258]
I like the idea of [this blogpost](http://www.dein.fr/2019-09-06-test-coverage-only-matters-if-at-100-percent.html). 
**TL;DR:** a project should have a test coverage of 100%. Not all lines should be tested, but lines we don't want to write a test for should be explicitly marked as such. We can ignore lines like this:
```ruby
# :nocov:
def skip_this_method
  never_reached
end
# :nocov:
```

I went trough our coverage report and we are currently just below 80%. There are some lines that we really want to cover (`institutions_controller.rb` is currently completely uncovered) but some are simply impossible to cover (`lib/SAML/my_resource_validator.rb` for example).

I suggest someone goes trough this report (found in `coverage/index.htm` after running `rails test`) and for each block of uncovered code:
- either writes a test covering that block
- either surrounds that block with `# :nocov: <Reason why this block is skipped>`

Finally, by using `SimpleCov.minimum_coverage 100` we can enforce that this being done for each line of extra code that is added to our project.

@bmesuere, @charvp what are your opinions? I think this would be an ideal issue for a job student.

**Comment by charvp** [comment]
Coverage reports are broken right now until https://github.com/colszowka/simplecov/issues/718 is fixed.

**Comment by rien** [comment]
Ah, I have a temporary fix on `feature/description-iframe`: https://github.com/dodona-edu/dodona/commit/2cebcbc48ea0235cfbd1bd24687b3265c8e3a691 which can be used.

Should I make a separate PR for this?

**Comment by charvp** [comment]
You can already set `PARALLEL_WORKERS=1` if you really want serial test execution.

**Comment by bmesuere** [comment]
+1 for this idea, but we must make sure that they are meaningful tests and not something to reach 100% coverage and be done with it

**Comment by rien** [comment]
> You can already set `PARALLEL_WORKERS=1` if you really want serial test execution.

Oh, nice. I'll remove that commit then.
